python/dgl/graphbolt/base.py

```diff
@@ -119,16 +121,25 @@ class CopyTo(IterDataPipe):
         The DataPipe.
     device : torch.device
         The PyTorch CUDA device.
+    extra_attrs: List[string]
+        The extra attributes in the MiniBatch you wish to be carried to the
+        specific device.
     """
 
-    def __init__(self, datapipe, device):
+    def __init__(self, datapipe, device, extra_attrs=None):
         super().__init__()
         self.datapipe = datapipe
         self.device = device
+        self.extra_attrs = extra_attrs
 
     def __iter__(self):
+        from .minibatch import MiniBatch
```

Comment:
Because it will cause circular importing

Reply:
if `apply_to(x, device, **kwargs)` and pass `kwargs` to `data.to()`, then we don't need to check instance type for `MiniBatch`. `Minibatch.to()` will handle the extra attrs?

Reply:
Because only `MiniBatch` has `to` function overridden to take and handle `extra_attrs`. Other classes can only accept `device` as the param for `to`. If we pass all the params it will error. But we can make `CopyTo` an `MiniBatchTransformer` so the data can only be `MiniBatch`.

Reply:
Actually, do we "CopyTo" any other class other than minibatch?

Reply:
If there is a user-defined stage in the data loader, it could be another class, tuple or anything.

Reply:
Then what does the UX would look like? In the GB datapipe api, I feel that the only contextual copy_to is for minibatch, other wise it will be very confusing.